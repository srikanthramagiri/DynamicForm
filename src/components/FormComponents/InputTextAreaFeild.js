import React, { useState } from "react";
import { Form, Row, Col } from "react-bootstrap";
import AlertError from "./AlertError";
import Label from "./Label";
import PropTypes from "prop-types"

export default function InputTextAreaFeild(props) {
  let {
    name,
    type,
    placeholder,
    label,
    text,
    changeHandler,
    rows,
    cols,
    validation,
  } = props;

  let { regex, errorMessage } = validation ? validation : {};
  let [isValidText, setIsValidText] = useState(true);
  let [areaText, setAreaText] = useState();

  let textAreaChangeHandler = (e) => {
    setIsValidText(true);
    setAreaText(e.target.value);
    changeHandler(e);
  };

  let textAreaBlurHandler = (e) => {
    if (e.target.value === "") {
      setIsValidText(false);
    }
    let regexValidation = new RegExp(regex);
    setIsValidText(regexValidation.test(e.target.value));
  };

  return (
    <Row>
      <Col lg={6} md={6}>
        <Label name={name} label={label} />
      </Col>
      <Col lg={6} md={6}>
        <Form.Control
          as="textarea"
          type={type}
          name={name}
          placeholder={placeholder}
          value={areaText}
          rows={rows}
          cols={cols}
          onChange={ (e) => textAreaChangeHandler(e)}
          onBlur={(e) => textAreaBlurHandler(e)}
        />
      </Col>
      <Col lg={6} md={6} />
      <Col lg={6} md={6}>
        {!isValidText ? (
          <AlertError variant={"danger"} message={errorMessage} />
        ) : (
          ""
        )}
      </Col>
    </Row>
  );
}

InputTextAreaFeild.prototype = {
  name: PropTypes.string,
    type: PropTypes.string,
    placeholder: PropTypes.string,
    label: PropTypes.string,
    text: PropTypes.string,
    changeHandler: PropTypes.func,
    rows:  PropTypes.number,
    cols: PropTypes.number,
    validation: PropTypes.object
}

InputTextAreaFeild.defaultProps = {
    rows:  20,
    cols: 40
}