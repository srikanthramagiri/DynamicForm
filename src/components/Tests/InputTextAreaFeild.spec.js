import { shallow } from 'enzyme';
import InputTextAreaFeild from '../FormComponents/InputTextAreaFeild'

describe('input text Area feild component', ()=> {
    const changeHandlerSpy = jest.fn();
    const wrapper = shallow(<InputTextAreaFeild changeHandler={changeHandlerSpy}/>)
    const InputTextArea =  wrapper.find(`[data-test='input-text-area-feild']`)
    it('should render correctly', ()=> {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).not.toBeNull()
    })
    it('should have one input text area',()=> {
        expect(InputTextArea).not.toBeNull()
    }) 
    it('change handler', ()=> {
        InputTextArea.simulate('change', {target: {value: 'ho.no.1-2-3'}})
        expect(changeHandlerSpy).toHaveBeenCalled()
    })
    it('blur handler ', ()=> {
        InputTextArea.simulate('blur', {target: {value: ''}})
        
    })
})