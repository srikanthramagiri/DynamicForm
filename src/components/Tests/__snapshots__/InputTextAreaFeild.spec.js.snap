// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input text Area feild component should render correctly 1`] = `ShallowWrapper {}`;

exports[`input text feild should render correctly 1`] = `ShallowWrapper {}`;
